mobile/src/full/java/org/openhab/habdroid/core/GcmRegistrationService.java

```diff
@@ -91,22 +88,13 @@ private void registerGcm(CloudConnection connection) throws IOException {
                 deviceId, deviceModel, token);
 
         Log.d(TAG, "Register device at openHAB-cloud with URL: " + regUrl);
-        connection.getSyncHttpClient().get(regUrl, new MyHttpClient.ResponseHandler() {
-            @Override
-            public void onFailure(Call call, int statusCode, Headers headers, byte[] responseBody, Throwable error) {
-                Log.e(TAG, "GCM reg id error: " + error.getMessage());
-                if (responseBody != null) {
-                    Log.e(TAG, "Error response = " + responseBody);
-                }
-                CloudMessagingHelper.sRegistrationFailureReason = error;
-            }
-
-            @Override
-            public void onSuccess(Call call, int statusCode, Headers headers, byte[] responseBody) {
-                Log.d(TAG, "GCM reg id success");
-                CloudMessagingHelper.sRegistrationFailureReason = null;
-            }
-        });
+        SyncHttpClient.HttpStatusResult result = connection.getSyncHttpClient().get(regUrl).asStatus();
+        if (result.error == null) {
```

Comment:
what do you think about a method for that, such "hasError" or even "isSuccessful"?

Reply:
Yeah, that's probably a good idea.